source/include/shadow.h

```diff
@@ -265,362 +307,342 @@ typedef enum ShadowStatus
+ *                                  - #SHADOW_OP_DELETE_LENGTH
+ *                                  - #SHADOW_OP_DELETE_ACCEPTED_LENGTH
+ *                                  - #SHADOW_OP_DELETE_REJECTED_LENGTH
+ *
  * @param[in] thingNameLength   Length of the thingName excluding the ending NULL.
  * @param[in] shadowNameLength  Length of the shadowName excluding the ending NULL. Zero for "Classic" shadow.
  *
  * @return Length of the shadow topic in bytes.
  */
-#define SHADOW_TOPIC_LENGTH( operationLength, suffixLength, thingNameLength, shadowNameLength ) \
-    ( operationLength + suffixLength + thingNameLength + shadowNameLength +                     \
-      SHADOW_PREFIX_LENGTH +                                                                    \
+#define SHADOW_TOPIC_LENGTH( operationLength, thingNameLength, shadowNameLength )   \
+    ( operationLength + thingNameLength + shadowNameLength + SHADOW_PREFIX_LENGTH + \
       ( ( shadowNameLength > 0 ) ? SHADOW_NAMED_ROOT_LENGTH : SHADOW_CLASSIC_ROOT_LENGTH ) )
 
 /**
- * @brief Compute the length of shadow topic "$aws/things/<thingName>/shadow/update" or
- *        "$aws/things/<thingName>/shadow/name/<shadowName>/update".
+ * @ingroup shadow_constants
+ * @brief Compute the length of the longest shadow topic.
  *
  * @param[in] thingNameLength   Length of the thingName excluding the ending NULL.
  * @param[in] shadowNameLength  Length of the shadowName excluding the ending NULL. Zero for "Classic" shadow.
  *
  * @return Length of the shadow topic in bytes.
  */
-#define SHADOW_TOPIC_LENGTH_UPDATE( thingNameLength, shadowNameLength ) \
-    SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_LENGTH, SHADOW_SUFFIX_NULL_LENGTH, thingNameLength, shadowNameLength )
+#define SHADOW_TOPIC_LENGTH_MAX( thingNameLength, shadowNameLength ) \
+    SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_DOCUMENTS_LENGTH, thingNameLength, shadowNameLength )
 
 /**
- * @brief Compute the length of shadow topic "$aws/things/<thingName>/shadow/update/accepted" or
- *        "$aws/things/<thingName>/shadow/name/<shadowName>/update/accepted".
+ * @ingroup shadow_constants
+ * @brief Assemble constant shadow topic strings when Thing Name and Shadow Name are known at compile time.
  *
- * @param[in] thingNameLength   Length of the thingName excluding the ending NULL.
- * @param[in] shadowNameLength  Length of the shadowName excluding the ending NULL. Zero for "Classic" shadow.
+ * When thingName and shadowName are known to be "myThing" and "myShadow" at compile time, invoke the macro like this:
  *
- * @return Length of the shadow topic in bytes.
+ *     SHADOW_TOPIC_STRING( SHADOW_OP_UPDATE_DELTA, "myThing", "myShadow" )
+ *
+ * When thingName and/or shadowName is only known at run time, do not use this macro. Use the
+ * Shadow_AssembleTopicString() function instead.
+ *
+ * @param[in] operation     Can be one of:
+ *                              - #SHADOW_OP_UPDATE
+ *                              - #SHADOW_OP_UPDATE_ACCEPTED
```

Comment:
I think it will be confusing to list SHADOW_OP_UPDATE_ACCEPTED as a possible value to the operation argument. Only GET, DELETE and UPDATE are Shadow operations. Rename the param name?

Reply:
Un-done by new commit today.